### PR TITLE
Upgrade GORM helper to use GORM v2

### PR DESCRIPTION
**Pull request by xuhui-lu**
Upgrade GORM helper to use GORM v2 #94

### CODE REVIEW

File: crdb/crdbgorm/gorm_test.go [hunk 1]
```diff
@@ -32,12 +35,12 @@ func TestExecuteTx(t *testing.T) {
 	}
 	ctx := context.Background()
 
-	gormDB, err := gorm.Open("postgres", ts.PGURL().String())
+	gormDB, err := gorm.Open(postgres.Open(ts.PGURL().String()), &gorm.Config{})
 	if err != nil {
 		t.Fatal(err)
 	}
 	// Set to true and gorm logs all the queries.
```
Comment (rafiss) [inline crdb/crdbgorm/gorm_test.go:42]: the comment should say "Set to logger.Info and gorm logs all the queries"

File: go.mod [hunk 2]
```diff
@@ -3,13 +3,18 @@ module github.com/cockroachdb/cockroach-go/v2
 go 1.13
 
 require (
-	github.com/jackc/pgx/v4 v4.6.0
-	github.com/jinzhu/gorm v1.9.12
+	github.com/jackc/pgx v3.6.2+incompatible // indirect
```
Comment (rafiss) [inline go.mod:6]: do you know why pgx v3 is added here? i don't think it should be

**Review by rafiss — CHANGES_REQUESTED** [review]
thanks for this update! i left a few comments

File: crdb/crdbgorm/gorm_test.go (hunk 1 above)
Comment (xuhui-lu) [reply crdb/crdbgorm/gorm_test.go]: thanks for the comment!

File: go.mod (hunk 2 above)
Comment (xuhui-lu) [reply go.mod:6]: yea, I agree. I've removed it from go.mod.

**Review by rafiss — APPROVED** [review]
looks great! thank you

**Comment by Amruta-Ranade** [comment]
@xuhui-lu Thank you for contributing to CockroachDB this year. As a token of our appreciation, we would like to send you a gift. Please DM me on our [community Slack](https://cockroa.ch/slack) @Amruta so I can send you a link. (If you don’t want a gift, we also have a charitable contribution choice.) All orders need to be in by December 13, so please contact me as soon as possible :)

**Pull request merged**